### ISSUE TITLE
Can't backward through a transformation matrix

### ISSUE THREAD

**Issue by simoncozens** [BachiLi/diffvg#101]
I'm guessing this project is pretty dead given the lack of updates, but filing this in case it bites someone else the way it bit me: if you're using a transformation matrix (the `shape_to_canvas` parameter to `pydiffvg.ShapeGroup`), gradients don't get transformed using the matrix. An obvious symptom of this is: if you flip the coordinate space upside down using a matrix, the updates to the shape will push it down when they should push it up and vice versa.

e.g. replacing

https://github.com/BachiLi/diffvg/blob/85802a71fbcc72d79cb75716eb4da4392fd09532/apps/single_curve.py#L24-L25

with

```python
transform = torch.tensor(
    [
        [1.0, 0.0, 0.0],
        [0.0, -1.0, 256.0],
        [0.0, 0.0, 1.0],
    ]
)
path_group = pydiffvg.ShapeGroup(
    shape_ids=torch.tensor([0]),
    fill_color=torch.tensor([0.3, 0.6, 0.3, 1.0]),
    shape_to_canvas=transform,
)
```

loss will increase, and the final result will look like this:

<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/eb6646b3-c223-4cfe-86e2-9302a7ce5494" />
